aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,199 @@
+
+Blazor Server benefits:
+
+* Create rich interactive UIs using [C#](/dotnet/csharp/) instead of [JavaScript](https://www.javascript.com).
+* Share server-side and client-side app logic written in .NET.
+* Render the UI as HTML and CSS for wide browser support, including mobile browsers.
+* Share app logic across server and client.
+* Leverage the existing .NET ecosystem of [.NET libraries](/dotnet/standard/class-libraries).
+* Flexibility of access to server resources, in a protected environment. For example:
+  * Database access.
+  * Access to secrets, such as an access key used for API calls to Azure storage.
+* Uses a token-based security model, rather than using cookies that may or may not be enabled on the browser.
+
+To get started with your first ASP.NET Core Blazor Server app, see [Get started with Blazor](https://dotnet.microsoft.com/learn/aspnet/blazor-tutorial/intro). For a complete overview of ASP.NET Core Blazor Server, its architecture and benefits, see [Introduction to ASP.NET Core Blazor](xref:blazor/index).
+
+### ASP.NET Core MVC plus hosted Blazor WebAssembly
+
+MVC and Blazor are both part of the ASP.NET Core framework and are designed to be used together. Razor components can be integrated into Razor Pages and MVC apps in a hosted Blazor WebAssembly solution. When the page or view is rendered, components can be prerendered at the same time.
+
+Choose ASP.NET Core MVC plus Blazor WebAssembly when these benefits best fit your Web UI app needs:
+
+* Prerendering:
+  * Executes Blazor components on the server and renders them into a page or view.
+	* Improves the perceived load time of the app while interactivity is being set up.
+* Add islands of interactivity to existing views (pages), using the component tag helper.
+
+To get started with ASP.NET Core MVC plus Blazor WebAssembly, see [Prerender and integrate ASP.NET Core Razor components](xref:blazor/components/prerendering-and-integration).
+
+### ASP.NET Core Single Page Application (SPA) with JavaScript Frameworks such as Angular and React
+
+Build client-side logic for ASP.NET Core apps using popular JavaScript frameworks, like [Angular](https://angular.io/) or [React](https://facebook.github.io/react/).  ASP.NET Core provides project templates for Angular, React, and React plus Redux.
+
+Benefits of ASP.NET Core SPA with JavaScript Frameworks, in addition to the client rendering benefits previously listed:
+
+* Client-side UI frameworks run on the user’s device in the browser
+* Client-based apps take advantage of the capabilities of the user’s device and provide rich interactivity.
+* Build client-side logic for ASP.NET Core apps using popular JS frameworks, like Angular and React, or build your client-side logic with .NET using Blazor.
+* Client-based apps can be deployed as standalone static sites, but often call into server hosted functionality through APIs.
+* Make your app installable and run offline by turning it into a Progressive Web App (PWA).
```

Comment:
These are all benefits of client-side web apps, but none of these appear to be specific to using JS. Instead I think we should list things like: 

- JS runtime included with the browser. No need to download one with your app.
- Large community and mature ecosystem

The downsides are that you have to now use a different language, different tools, different frameworks, etc. You can't really share code, so logic often needs to get duplicated.